src/state/routing/slice.ts

```diff
@@ -23,9 +23,34 @@ const baseQuery: BaseQueryFn<GetQuoteArgs, TradeResult> = () => {
 export const routing = createApi({
   reducerPath: 'routing',
   baseQuery,
-  serializeQueryArgs: serializeGetQuoteArgs,
+  serializeQueryArgs: serializeGetQuoteQueryArgs,
   endpoints: (build) => ({
     getTradeQuote: build.query({
+      async onQueryStarted(args: GetQuoteArgs | SkipToken, { queryFulfilled }) {
+        if (args === skipToken) return
+
+        args.onQuote?.(
+          JSON.parse(serializeGetQuoteArgs(args)),
+          queryFulfilled
+            .catch((error) => {
+              const { error: queryError } = error
+              if (queryError && typeof queryError == 'object' && 'status' in queryError) {
+                const parsedError = queryError as FetchBaseQueryError
+                switch (parsedError.status) {
+                  case 'CUSTOM_ERROR':
+                  case 'FETCH_ERROR':
+                  case 'PARSING_ERROR':
+                    throw parsedError.error
+                    break
+                  default:
+                    throw parsedError.status
+                }
+              }
+              throw queryError
```

Comment:
is there a way to indicate in `perf.ts` that these can throw certain types of errors? should the integrator be aware so they can handle them?

Reply:
I think a good follow up would be for _us_ to export an enum (or extend `Promise`) and only throw these promises for integration hooks, but I don't think that that should be a part of this PR. It could be a future improvement.

Reply:
sounds good - i was imagining some generic "throws" descriptor on the function , not even typing it. maybe something worth calling out in the docs if we update them for the PerfEventHandlers